src/test/java/teammates/ui/webapi/GetDeadlineExtensionActionTest.java

```diff
@@ -0,0 +1,79 @@
+package teammates.ui.webapi;
+
+import java.time.Instant;
+
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.attributes.DeadlineExtensionAttributes;
+import teammates.common.util.Const;
+import teammates.ui.output.DeadlineExtensionData;
+
+/**
+ * SUT: {@link GetDeadlineExtensionsAction}.
+ */
+public class GetDeadlineExtensionActionTest extends BaseActionTest<GetDeadlineExtensionAction> {
+
+    @Override
+    protected String getActionUri() {
+        return Const.ResourceURIs.DEADLINE_EXTENSION;
+    }
+
+    @Override
+    protected String getRequestMethod() {
+        return GET;
+    }
+
+    @Override
+    @Test
+    protected void testExecute() {
+        DeadlineExtensionAttributes deadlineExtension = typicalBundle.deadlineExtensions.get("student4InCourse1Session1");
+
+        loginAsAdmin();
+
+        ______TS("Not enough parameters");
+
+        verifyHttpParameterFailure();
+
+        ______TS("deadline extension does not exist");
+
+        String[] nonExistParams = {
+                Const.ParamsNames.COURSE_ID, "unknown-course-id",
+                Const.ParamsNames.FEEDBACK_SESSION_NAME, "unknown-fs-name",
+                Const.ParamsNames.USER_EMAIL, "unknown@gmail.tmt",
+                Const.ParamsNames.IS_INSTRUCTOR, "false",
+        };
+
+        EntityNotFoundException enfe = verifyEntityNotFound(nonExistParams);
+        assertEquals("Deadline extension for course id: unknown-course-id and "
+                + "feedback session name: unknown-fs-name and student email: unknown@gmail.tmt not found.",
+                enfe.getMessage());
+
+        ______TS("typical success case");
+
+        String[] params = new String[] {
+                Const.ParamsNames.COURSE_ID, deadlineExtension.getCourseId(),
+                Const.ParamsNames.FEEDBACK_SESSION_NAME, deadlineExtension.getFeedbackSessionName(),
+                Const.ParamsNames.USER_EMAIL, deadlineExtension.getUserEmail(),
+                Const.ParamsNames.IS_INSTRUCTOR, Boolean.toString(deadlineExtension.getIsInstructor()),
+        };
+
+        GetDeadlineExtensionAction a = getAction(params);
+        JsonResult r = getJsonResult(a);
+
+        DeadlineExtensionData response = (DeadlineExtensionData) r.getOutput();
+
+        assertEquals(deadlineExtension.getCourseId(), response.getCourseId());
+        assertEquals(deadlineExtension.getFeedbackSessionName(), response.getFeedbackSessionName());
+        assertEquals(deadlineExtension.getUserEmail(), response.getUserEmail());
+        assertEquals(deadlineExtension.getIsInstructor(), response.getIsInstructor());
```

Comment:
For ID fields we can use `.equals`?

Reply:
Are you suggesting using `assertTrue(x.equals(y))`?

Reply:
At least for identifier fields, we can do so? Not sure if this makes the code less readable.

Reply:
This is possible for the first 3 (except `isInstructor` which is a primitive boolean value). 

Implementation wise, it doesn't make much of a difference, except that `assertEquals` is null safe although that doesn't come into play here. Both use `.equals` to compare objects.

Comes down to readability I think. Personally prefer `assertEquals` here because it
1. Gives better error message when condition fails
2. Makes it explicit that we are checking for equality

What's the rationale behind using `assertTrue(x.equals(y))` instead?

Reply:
I see. I do feel that the readability of the test failure will be worse.

The rationale is to compare id fields then other fields, as id fields should be the same regardless?

Reply:
Sorry I don't see the difference here. This tests that "id fields" are the same as well? There shouldn't be any difference between `assertEquals` and `.equals` for non-null fields, since they both use `.equals` to compare. Are you referring to comparing identity using `==` instead?

Reply:
I think nvm... The current way is more straightforward.